### ISSUE TITLE
Bookmarks are not appearing immediately

### ISSUE THREAD

**Issue by FaroukZeino** [insidegui/WWDC#553]
When adding a bookmark to a streaming video, it's not appearing immediately, I need to play the video for a couple of seconds to get it appears.
I am not sure of downloaded videos, but I think they have the same problem.

**Comment by allenhumphreys** [comment]
Do you mean "favorite"?

**Comment by FaroukZeino** [comment]
No, have a look [https://imgur.com/a/KSTqq66](here)

**Comment by allenhumphreys** [comment]
Sorry I was confused. Could you upload a brief gif or screen recording of the behavior by chance?

**Comment by allenhumphreys** [comment]
The delay has to do with generating a thumbnail image we use for the UI. It's likely way more pronounced when using when playing a streaming video compared to a downloaded video (generating a thumbnail will be dependent on internet connection as well). To make the UI snappier, the architecture would need to change somewhat to allow for a deferred delivery of a video thumbnail after the creation of the bookmark.

For reference:
<img width="1911" alt="Screen Shot 2019-06-05 at 9 31 15 PM" src="https://user-images.githubusercontent.com/8225090/59002818-5417a700-87d9-11e9-809a-de9a2100aace.png">

**Comment by FaroukZeino** [comment]
I still don't understand the relation between downloading the thumbnail and creating a bookmark? can't the thumbnail created after the bookmark?

**Comment by FaroukZeino** [comment]
I have uploaded a video [here](https://imgur.com/1fcytxs) to show you what I exactly mean, it took about 14 seconds for the bookmark icon to appear.

**Comment by allenhumphreys** [comment]
Yeah. This is a valid issue. The program flow seems to optimize for the downloaded videos, but it does work as implemented.

**Comment by insidegui** [comment]
Yeah we need to implement lazy loading of thumbnails for bookmarks. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
